### PR TITLE
Update pzemac.rst due to wrong PIN reference

### PR DESCRIPTION
**Pull request by emeggiolaro**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by jesserockz — APPROVED** [review]
Example pins are just examples. A lot of boards and devices have different pins for different functions. 
All though I do somewhat agree that maybe we should use the `GPIOx` variant everywhere and not the board specific pins `Dx` etc.

@glmnet Thoughts? Should we enforce `GPIOx` instead of `Dx` for docs?

**Comment by glmnet** [comment]
yes, I'd rather use `GPIOxx` instead of `Dx` however this PR is not replacing just PIN names, it's actually using default RX TX pins (on both ESP8266 and ESP32) which are not default in given example, in this case I would just remove the `TX:` `RX:` config lines as they are default and we do not put defaults in examples.

Using default uart pins will mean you needing to disable serial logging, so this PR should remove RX and TX and add `baud_rate: 0` to `logger:`, otherwise the sample is OK

RELATED: We could add comments indicating `# ESP8266` or `# ESP32` pin setup, most lovely would be to have a button to view samples for one device or another like some tutorials sites they show you javascript / typescript etc.

File: components/sensor/pzemac.rst [hunk 1]
```diff
@@ -35,8 +35,8 @@ to some pins on your board and the baud rate set to 9600.
 
     # Example configuration entry
     uart:
-      rx_pin: D1
-      tx_pin: D2
+      rx_pin: GPIO3
+      tx_pin: GPIO1
```
Comment (frenck) [inline components/sensor/pzemac.rst:39]:
```suggestion
```

File: components/sensor/pzemac.rst [hunk 2]
```diff
@@ -35,8 +35,8 @@ to some pins on your board and the baud rate set to 9600.
...
       baud_rate: 9600
```
Comment (frenck) [inline components/sensor/pzemac.rst:40]:
```suggestion
      baud_rate: 9600
      
    logger:
      baud_rate: 0
```

**Review by frenck — CHANGES_REQUESTED** [review]
Added 2 review suggestions to this PR, as per review comments by @glmnet.

@emeggiolaro, could you take a look? Thanks already! 👍 


../Frenck 🚂

**Pull request closed without merging**